src/sentry/api/endpoints/organization_teams.py

```diff
@@ -70,15 +75,29 @@ def get(self, request, organization):
         if request.auth and hasattr(request.auth, 'project'):
             return Response(status=403)
 
-        team_list = list(
-            Team.objects.filter(
-                organization=organization,
-                status=TeamStatus.VISIBLE,
-            ).order_by('name', 'slug')
+        queryset = Team.objects.filter(
+            organization=organization,
+            status=TeamStatus.VISIBLE,
+        ).order_by('slug')
+
+        query = request.GET.get('query')
+        if query:
+            tokens = tokenize_query(query)
+            for key, value in six.iteritems(tokens):
+                if key == 'query':
+                    value = ' '.join(value)
+                    queryset = queryset.filter(Q(name__icontains=value) | Q(slug__icontains=value))
+                else:
+                    queryset = queryset.none()
+
+        return self.paginate(
+            request=request,
+            queryset=queryset,
+            order_by='slug',
+            on_results=lambda x: serialize(x, request.user, TeamWithProjectsSerializer()),
```

Comment:
note: we almost certainly want to follow up and tweak this serializer since its really heavy now